### PR TITLE
feat: add lease client

### PR DESCRIPTION
**Pull request by LingKa28**
Base on #3 

Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)
  add new feature

* what changes does this pull request make?
  * add lease client & test samples

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)
  no

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -110,3 +123,32 @@ func newClientTimeout(options ClientTimeout) ClientTimeout {
...
+		panic(err)
+	}
+
+	var id uint64
+	atomic.StoreUint64(&id, binary.BigEndian.Uint64(b[:]))
+	return leaseIdGenerator{id: id}
+}
+
+// Generate a new `leaseId`
+func (g leaseIdGenerator) next() int64 {
```
Comment (iGxnon) [inline client/client.go:148]: Should it be `func (g *leaseIdGenerator)` here?

File: client/client.go [hunk 2]
```diff
@@ -110,3 +123,32 @@ func newClientTimeout(options ClientTimeout) ClientTimeout {
...
+func newLeaseId() leaseIdGenerator {
+	var b [8]byte
+	_, err := rand.Read(b[:])
+	if err != nil {
+		panic(err)
+	}
+
+	var id uint64
+	atomic.StoreUint64(&id, binary.BigEndian.Uint64(b[:]))
+	return leaseIdGenerator{id: id}
```
Comment (iGxnon) [inline client/client.go:144]: It seems just generate a random u64 here. Use `rand.Intn` is sufficient.

File: client/client.go [hunk 3]
```diff
@@ -110,3 +123,32 @@ func newClientTimeout(options ClientTimeout) ClientTimeout {
...
+
+func newLeaseId() leaseIdGenerator {
+	var b [8]byte
+	_, err := rand.Read(b[:])
+	if err != nil {
+		panic(err)
+	}
+
+	var id uint64
+	atomic.StoreUint64(&id, binary.BigEndian.Uint64(b[:]))
```
Comment (iGxnon) [inline client/client.go:143]: We don't need atomic operations here because `id` can be only accessed in this context.

File: client/lease.go [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+		curpClient:  curpClient,
+		leaseClient: innerLeaseClient,
+		token:       token,
+		idGen:       idGen,
+	}
+}
+
+// Build inner client
+func buildLeaseClientFromAddrs(addrs []string, timeout ClientTimeout) (xlineapi.LeaseClient, error) {
+	conn, err := grpc.Dial(addrs[0], grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithIdleTimeout(timeout.idleTimeout))
```
Comment (iGxnon) [inline client/lease.go:46]: You can add load balancing to `addrs`. These changes can be updated in subsequent PRs.

File: client/client.go [hunk 5]
```diff
@@ -110,3 +123,32 @@ func newClientTimeout(options ClientTimeout) ClientTimeout {
 
 	return ct
 }
+
+// / Generator of unique lease id
```
Comment (Phoenix500526) [inline client/client.go:127]: Please remove this useless slash.

**Review by Phoenix500526 — CHANGES_REQUESTED** [review]

File: client/client.go (hunk 1 above)
Comment (LingKa28) [reply client/client.go:148]: Yes, it should be. I have followed your suggestion to modify it.

File: client/client.go (hunk 2 above)
Comment (LingKa28) [reply client/client.go:144]: You're right. I have followed your suggestion to modify it.

File: client/lease.go (hunk 4 above)
Comment (LingKa28) [reply client/lease.go:46]: Okay, I'll update this in subsequent PRs.

File: client/client.go (hunk 5 above)
Comment (LingKa28) [reply client/client.go:127]: Oops, sorry I missed that! I've updated it as you recommended.

**Review by Phoenix500526 — APPROVED** [review]
LGTM

**Review by iGxnon — APPROVED** [review]
LGTM

**Pull request merged**